tests/raftstore/transport_simulate.rs

```diff
@@ -293,54 +294,35 @@ impl Filter<RaftMessage> for FilterRegionPacket {
     }
 }
 
-pub struct IsolateRegionStore {
-    region_id: u64,
-    store_id: u64,
-    direction: Direction,
-    allow: usize,
-    msg_type: Option<MessageType>,
-}
-
-impl IsolateRegionStore {
-    pub fn new(region_id: u64, store_id: u64) -> IsolateRegionStore {
-        IsolateRegionStore {
+impl FilterRegionPacket {
+    pub fn new(region_id: u64, store_id: u64) -> FilterRegionPacket {
+        FilterRegionPacket {
             region_id: region_id,
             store_id: store_id,
             direction: Direction::Both,
             msg_type: None,
-            allow: 0,
+            allow: Arc::new(AtomicUsize::new(0)),
         }
     }
 
-    pub fn direction(mut self, direction: Direction) -> IsolateRegionStore {
+    pub fn direction(mut self, direction: Direction) -> FilterRegionPacket {
         self.direction = direction;
         self
     }
 
-    pub fn msg_type(mut self, m_type: MessageType) -> IsolateRegionStore {
+    pub fn msg_type(mut self, m_type: MessageType) -> FilterRegionPacket {
         self.msg_type = Some(m_type);
         self
     }
 
-    pub fn allow(mut self, number: usize) -> IsolateRegionStore {
-        self.allow = number;
+    pub fn allow(mut self, number: usize) -> FilterRegionPacket {
+        self.allow = Arc::new(AtomicUsize::new(number));
         self
     }
 }
 
-impl FilterFactory for IsolateRegionStore {
-    fn generate(&self, _: u64) -> Vec<Box<Filter<RaftMessage>>> {
-        vec![box FilterRegionPacket {
-                 region_id: self.region_id,
-                 store_id: self.store_id,
-                 direction: self.direction,
-                 msg_type: self.msg_type.clone(),
-                 allow: AtomicUsize::new(self.allow),
-             }]
-    }
-}
-
-struct SnapshotFilter {
+#[derive(Default)]
+pub struct SnapshotFilter {
```

Comment:
xxxFilter or Filterxxx? Both occur in this file.
IMO, xxxFilter would be a better name for a struct. And may every filter factory should has the suffix 'FilterFactory'.
I think it needs to be uniformed for the naming convention. Maybe we could do it some time later.